File: apps/byteport/sdk/typescript/examples/stream-logs.ts
/**
 * Stream logs example
 */

import { BytePortClient } from '../src'

async function main() {
  const apiKey = process.env.BYTEPORT_API_KEY
  if (!apiKey) {
    console.error('Error: BYTEPORT_API_KEY environment variable not set')
    process.exit(1)
  }

  const deploymentId = process.argv[2]
  if (!deploymentId) {
    console.error('Usage: node stream-logs.js <deployment_id>')
    process.exit(1)
  }

  const client = new BytePortClient({ apiKey })

  console.log(`📜 Streaming logs for deployment: ${deploymentId}`)
  console.log('Press Ctrl+C to stop\n')

  try {
    for await (const log of client.streamLogs(deploymentId)) {
      const timestamp = new Date(log.timestamp).toLocaleTimeString()
      console.log(`[${timestamp}] [${log.level}] ${log.message}`)
    }
  } catch (error) {
    console.error('Error streaming logs:', error)
    process.exit(1)
  }
}

main()

File: apps/byteport/sdk/go/examples/stream_logs.go
package main

import (
	"context"
	"fmt"
	"log"
	"os"
	"os/signal"
	"syscall"

	"github.com/byteport/sdk-go/byteport"
)

func main() {
	apiKey := os.Getenv("BYTEPORT_API_KEY")
	if apiKey == "" {
		log.Fatal("BYTEPORT_API_KEY environment variable not set")
	}

	if len(os.Args) < 2 {
		log.Fatal("Usage: stream_logs <deployment_id>")
	}

	deploymentID := os.Args[1]
	client := byteport.NewClient(apiKey)

	// Create context with cancellation
	ctx, cancel := context.WithCancel(context.Background())
	defer cancel()

	// Handle interrupt signal
	sigChan := make(chan os.Signal, 1)
	signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM)
	go func() {
		<-sigChan
		fmt.Println("\n\n🛑 Stopping log stream...")
		cancel()
	}()

	fmt.Printf("📜 Streaming logs for deployment: %s\n", deploymentID)
	fmt.Println("Press Ctrl+C to stop\n")

	// Stream logs
	logChan, errChan, err := client.StreamLogs(ctx, deploymentID)
	if err != nil {
		log.Fatalf("Failed to stream logs: %v", err)
	}

	// Process logs
	for {
		select {
		case logEntry, ok := <-logChan:
			if !ok {
				fmt.Println("\n✅ Log stream ended")
				return
			}
			fmt.Printf("[%s] [%s] %s\n",
				logEntry.Timestamp.Format("15:04:05"),
				logEntry.Level,
				logEntry.Message)

		case err, ok := <-errChan:
			if ok && err != nil {
				log.Printf("❌ Stream error: %v", err)
			}
			return

		case <-ctx.Done():
			return
		}
	}
}
